testing/citest/tests/kube_smoke_test.py

```diff
@@ -235,6 +234,159 @@ def delete_load_balancer(self):
+
+  def create_daemonset_pipeline(self):
+    bindings = self.bindings
+    payload = self.agent.make_json_payload_from_kwargs(
+        job=[{
+            'type': 'manual',
+            'user': '[anonymous]'
+        }],
+        description='Create Daemonset'.format(
+            self.__lb_name,
+            bindings['SPINNAKER_KUBERNETES_ACCOUNT']),
+        application=self.TEST_APP)
+    builder = kube.KubeContractBuilder(self.kube_observer)
+    (builder.new_clause_builder('Run pipeline', retryable_for_secs=15)
+     .get_resources(
+         'svc',
+         extra_args=['--namespace', self.TEST_NAMESPACE],
+         no_resource_ok=True))
+    return st.OperationContract(
+        self.new_post_operation(
+            title='Create_daemonset', data=payload, path='pipelines/'+self.TEST_APP+'/daemonset-pipeline',
+            status_class=st.SynchronousHttpOperationStatus),
+        contract=builder.build())
+
+
+  def save_daemonset_pipeline(self):
+      bindings = self.bindings
+     # import pdb;pdb.set_trace();
+      payload = self.agent.make_json_payload_from_kwargs(
+			    job=[{
+	                'keepWaitingPipelines': 'false',
+                        'application': self.TEST_APP,
+                        'name': 'daemonset-pipeline',
+	                'lastModifiedBy': 'anonymous',
+	                'limitConcurrent': 'true',
+	                'parallel': 'true',
+	                'stages': [{
+		                'clusters': [{
+			                'account': bindings['SPINNAKER_KUBERNETES_ACCOUNT'],
+			                'application': self.TEST_APP,
+			                'cloudProvider': 'kubernetes',
+			                'containers': [{
+				                    'imageDescription': {
+					                'repository': 'nginx'
+				                        },
+				            'imagePullPolicy': 'ALWAYS',
+				            'limits': {
+					            'memory': '200Mi'
+				                    },
+				            'name': 'nginx',
+				            'requests': {
+					            'cpu': '100m',
+					            'memory': '200Mi'
+				                }
+			                }],
+			        'kind': 'DaemonSet',
+			        'namespace': 'default',
+			        'provider': 'kubernetes',
+			        'region': 'default',
+			        'targetSize': 0,
+			        'terminationGracePeriodSeconds': 30
+		                }],
+		                'name': 'Deploy',
+		                'refId': '1',
+
+		            'type': 'deploy'
+	                }],
+	                'triggers': [{'enabled': 'true'}]
+                }],
+                    description='Save Daemonset Pipenine:' ,application=self.TEST_APP)
+      builder = kube.KubeContractBuilder(self.kube_observer)
+      builder.new_clause_builder('Run pipeline', retryable_for_secs=15).get_resources('svc',extra_args=['--namespace', self.TEST_NAMESPACE],no_resource_ok=True)
+      my_dict = eval(payload)
```

Comment:
I am confused. You are never using the json payload as json.
Instead "my_dict" is just the dictionary you built up to pass into the make_json_payload_from_kwargs, so just build up the dictionary.
Furthermore, you are just extracting 'job' out, so just set a job variable here to the dictionary you want. But then you convert it to a string and change the quoting syntax then pass it as the data to the post operation. So I assume it is in fact supposed to be a JSON encoding.

So really I think the "payload" above should just be the job dictionary, then return here with data=payload. Or call json.JSONEncoder().encode directly with the job dictionary if you'd prefer to specify the payload as a dictionary given its complexity.